### PR TITLE
THIRDEYE-876 : Config driven support in thirdeye dashboard for tables having metrics as a dimension column

### PR DESCRIPTION
**Pull request by npawar**
This pull request adds support for tables which have metric names as a dimension column, and metric values as a single metric column. It handles derived metrics as well.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/pinot/PinotThirdEyeClient.java [hunk 1]
```diff
@@ -109,18 +109,46 @@ public PinotThirdEyeResponse execute(ThirdEyeRequest request) throws Exception {
     TimeSpec dataTimeSpec = collectionSchema.getTime();
     List<MetricFunction> metricFunctions = request.getMetricFunctions();
     List<String> dimensionNames = collectionSchema.getDimensionNames();
-    String sql = PqlUtils.getPql(request, dataTimeSpec);
-    LOG.debug("Executing: {}", sql);
-    ResultSetGroup result =
-        CACHE_INSTANCE.getResultSetGroupCache().get(
-            new PinotQuery(sql, request.getCollection() + "_OFFLINE"));
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/pinot/PinotThirdEyeClient.java]: why should we append _OFFLINE, does this mean it wont work with realtime table?

**Comment by kishoreg** [comment]
everything else looks good, can you clarify if this works for both REALTIME/OFFLINE tables

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/pinot/PinotThirdEyeClient.java (hunk 1 above)
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/pinot/PinotThirdEyeClient.java]: Created THIRDEYE-877 to address this

**Comment by npawar** [comment]
Testing done: Onboarded internal datasets where metric name is a single column, and metric value is a single column. Tested that the dashboard can handle all these metrics as separate metrics and process/display them as separate entities. Confirmed that derived metrics is also handled in these cases

**Pull request merged**